tests/test_views.py

```diff
@@ -1,85 +1,82 @@
 from django.contrib.contenttypes.models import ContentType
 from django.contrib.sites.models import Site
-from django.test import RequestFactory
+from django.urls.exceptions import NoReverseMatch
 
 from cms.models import Page, User
 from cms.test_utils.testcases import CMSTestCase
 from cms.utils.urlutils import admin_reverse
 
 from djangocms_navigation.constants import SELECT2_CONTENT_OBJECT_URL_NAME
+from djangocms_navigation.models import MenuItem
 from djangocms_navigation.test_utils.factories import (
     MenuContentFactory,
     PageContentFactory,
-    UserFactory,
 )
 from djangocms_navigation.test_utils.polls.models import Poll, PollContent
-from djangocms_navigation.views import MenuContentPreviewView
 
 
-class PreviewViewTestCases(CMSTestCase):
+
+class PreviewViewPermissionTestCases(CMSTestCase):
     def setUp(self):
         self.menu_content = MenuContentFactory()
         self.preview_url = admin_reverse(
             "djangocms_navigation_menuitem_preview",
             kwargs={"menu_content_id": self.menu_content.id},
         )
 
-    def test_view_url(self):
-        expcted_url = "/en/admin/djangocms_navigation/menuitem/{}/preview/".format(
-            self.menu_content.id
-        )
-        self.assertEqual(self.preview_url, expcted_url)
-
-    def test_view_anonymous_user(self):
+    def test_anonymous_user_cannot_access_preview(self):
         response = self.client.get(self.preview_url)
         expected_url = "/en/admin/login/?next=" + self.preview_url
         self.assertEqual(response.status_code, 302)
         self.assertEqual(response.url, expected_url)
 
-    def test_view_standard_user(self):
+    def test_standard_user_cannot_access_preview(self):
         standard_user = self.get_standard_user()
         with self.login_user_context(standard_user):
             response = self.client.get(self.preview_url)
             expected_url = "/en/admin/login/?next=" + self.preview_url
             self.assertEqual(response.status_code, 302)
             self.assertEqual(response.url, expected_url)
 
-    def test_view_super_user(self):
-        staff_user = self.get_superuser()
-        with self.login_user_context(staff_user):
-            response = self.client.get(self.preview_url)
-            self.assertEqual(response.status_code, 200)
+
+class PreviewViewTestCases(CMSTestCase):
+    def setUp(self):
+        self.menu_content = MenuContentFactory()
```

Comment:
Correct me if I'm wrong but I think you're only using this db object in just one test in this test case, so makes more sense to define it there rather than in `setUp`.